protocol/lavasession/provider_session_manager.go

```diff
@@ -168,10 +169,24 @@ func (psm *ProviderSessionManager) GetSession(ctx context.Context, address strin
 		return nil, err
 	}
 
-	return psm.getSingleSessionFromProviderSessionWithConsumer(ctx, providerSessionsWithConsumer, sessionId, epoch, relayNumber)
+	return psm.getSingleSessionFromProviderSessionWithConsumer(ctx, providerSessionsWithConsumer, sessionId, epoch, relayNumber, badgeSession)
 }
 
-func (psm *ProviderSessionManager) registerNewConsumer(consumerAddr string, epoch uint64, maxCuForConsumer uint64, pairedProviders int64) (*ProviderSessionsWithConsumer, error) {
+func getBadgeEpochDataFromProviderSessionWithConsumer(badgeUser string, providerSessionsWithConsumer *ProviderSessionsWithConsumer) (*ProviderSessionsEpochData, bool) {
+	providerSessionsWithConsumer.Lock.RLock()
+	defer providerSessionsWithConsumer.Lock.RUnlock()
+	badgeUserEpochData, exists := providerSessionsWithConsumer.badgeEpochData[badgeUser]
+	return badgeUserEpochData, exists
+}
+
+func registerBadgeEpochDataToProviderSessionWithConsumer(badgeUser string, badgeCuAllocation uint64, providerSessionsWithConsumer *ProviderSessionsWithConsumer) *ProviderSessionsEpochData {
+	providerSessionsWithConsumer.Lock.Lock()
+	defer providerSessionsWithConsumer.Lock.Unlock()
+	providerSessionsWithConsumer.badgeEpochData[badgeUser] = &ProviderSessionsEpochData{MaxComputeUnits: uint64(math.Min(float64(providerSessionsWithConsumer.epochData.MaxComputeUnits), float64(badgeCuAllocation)))}
```

Comment:
do not type convert here just because there is no min max function for uint, please use a swtich case in a helper function (we already have one implemented in the repo can't remember it's location look for it or ask yarom, so just use it)

Reply:
I wasn't able to find existing implementation so I have added a small function that serves the purpose: 65ac1e43431a0f06114af49c6730270e511e1f8b